File: backend/BudgetManagementAPI/Models/User/UserDetailsDto.cs
using BudgetManagementAPI.Enums;

namespace BudgetManagementAPI.Models.User;

public class UserDetailsDto
{
    public Guid UserId { get; set; }
    public required string Name { get; set; }
    public required string Email { get; set; }
    public required double Income { get; set; }
    public required Profession Profession {  get; set; }
    public required Level LevelFinancialEducation { get; set; }
}

File: budget-management-app/src/app/models/userDetailsDto.ts
import { Currency } from '../enums/currency';
import { CurrencyDisplay } from '../enums/currencyDisplay';
import { LevelDisplay } from '../enums/levelDisplay';
import { ProfessionDisplay } from '../enums/professionDisplay';

export class UserDetailsDto {
  UserId: string = '';
  Name: string = '';
  Email: string = '';
  Income: number = 0;
  Profession!: ProfessionDisplay;
  LevelFinancialEducation!: LevelDisplay;
  Currency!: Currency;
}
